### PR TITLE
feat: add DuckDB `Struct` dtype

### PR DESCRIPTION
**Pull request by raisadz**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #1728
- Closes #\<issue number\>

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_duckdb/utils.py [hunk 1]
```diff
@@ -157,9 +157,9 @@ def native_to_narwhals_dtype(duckdb_dtype: str, version: Version) -> DType:
 def narwhals_to_native_dtype(dtype: DType | type[DType], version: Version) -> str:
     dtypes = import_dtypes_module(version)
     if isinstance_or_issubclass(dtype, dtypes.Float64):
-        return "FLOAT"
-    if isinstance_or_issubclass(dtype, dtypes.Float32):
         return "DOUBLE"
+    if isinstance_or_issubclass(dtype, dtypes.Float32):
+        return "FLOAT"
     if isinstance_or_issubclass(dtype, dtypes.Int64):
```
Comment (MarcoGorelli) [inline narwhals/_duckdb/utils.py:163]: 😳 oops, nice catch!

File: narwhals/_duckdb/utils.py [hunk 2]
```diff
@@ -198,8 +198,11 @@ def narwhals_to_native_dtype(dtype: DType | type[DType], version: Version) -> st
         inner = narwhals_to_native_dtype(dtype.inner, version)  # type: ignore[union-attr]
         return f"{inner}[]"
     if isinstance_or_issubclass(dtype, dtypes.Struct):  # pragma: no cover
-        msg = "todo"
-        raise NotImplementedError(msg)
+        inner = ", ".join(
+            f"{field.name} {narwhals_to_native_dtype(field.dtype, version)}"
+            for field in dtype.fields  # type: ignore[union-attr]
+        )
+        return f"STRUCT({inner})"
```
Comment (MarcoGorelli) [inline narwhals/_duckdb/utils.py:205]: does this work if column names contain spaces? we might need
```
f'"{field.name}" {narwhals_to_native_dtype(field.dtype, version)}'
```
?

**Review by MarcoGorelli — APPROVED** [review]
amazing, thanks @raisadz !

File: narwhals/_duckdb/utils.py (hunk 2 above)
Comment (raisadz) [reply narwhals/_duckdb/utils.py]: No, it didn't work, thanks for pointing this out! I added the suggestion and also changed the test to have a column with an empty space
